edn/src/edn.rustpeg

```diff
@@ -34,125 +34,54 @@ use types::{SpannedValue, Span, ValueAndSpan};
-        ValueAndSpan {
-            inner: SpannedValue::Nil,
-            span: Span::new(start, end)
-        }
-    }
+pub nil -> SpannedValue = "nil" { SpannedValue::Nil }
+pub nan -> SpannedValue = "#f" whitespace+ "NaN" { SpannedValue::Float(OrderedFloat(NAN)) }
 
-pub nan -> ValueAndSpan =
-    start:#position "#f" whitespace+ "NaN" end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::Float(OrderedFloat(NAN)),
-            span: Span::new(start, end)
-        }
-    }
-
-pub infinity -> ValueAndSpan =
-    start:#position "#f" whitespace+ s:$(sign) "Infinity" end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::Float(OrderedFloat(if s == "+" { INFINITY } else { NEG_INFINITY })),
-            span: Span::new(start, end)
-        }
-    }
+pub infinity -> SpannedValue = "#f" whitespace+ s:$(sign) "Infinity"
+    { SpannedValue::Float(OrderedFloat(if s == "+" { INFINITY } else { NEG_INFINITY })) }
 
-pub boolean -> ValueAndSpan =
-    start:#position "true" end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::Boolean(true),
-            span: Span::new(start, end)
-        }
-    } /
-    start:#position "false" end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::Boolean(false),
-            span: Span::new(start, end)
-        }
-    }
+pub boolean -> SpannedValue
+    = "true"  { SpannedValue::Boolean(true) }
+    / "false" { SpannedValue::Boolean(false) }
 
 digit = [0-9]
 alphanumeric = [0-9a-zA-Z]
 octaldigit = [0-7]
 validbase = [3][0-6] / [12][0-9] / [2-9]
 hex = [0-9a-fA-F]
-sign = "-" / "+"
+sign = [+-]
 
-pub bigint -> ValueAndSpan =
-    start:#position b:$( sign? digit+ ) "N" end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::BigInteger(b.parse::<BigInt>().unwrap()),
-            span: Span::new(start, end)
-        }
-    }
+pub bigint -> SpannedValue = b:$( sign? digit+ ) "N"
+    { SpannedValue::BigInteger(b.parse::<BigInt>().unwrap()) }
+pub octalinteger -> SpannedValue = "0" i:$( octaldigit+ )
+    { SpannedValue::Integer(i64::from_str_radix(i, 8).unwrap()) }
+pub hexinteger -> SpannedValue = "0x" i:$( hex+ )
+    { SpannedValue::Integer(i64::from_str_radix(i, 16).unwrap()) }
 
-pub octalinteger -> ValueAndSpan =
-    start:#position "0" i:$( octaldigit+ ) end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::Integer(i64::from_str_radix(i, 8).unwrap()),
-            span: Span::new(start, end)
-        }
-    }
+pub basedinteger -> SpannedValue = b:$( validbase ) "r" i:$( alphanumeric+ )
+    { SpannedValue::Integer(i64::from_str_radix(i, b.parse::<u32>().unwrap()).unwrap()) }
 
-pub hexinteger -> ValueAndSpan =
-    start:#position "0x" i:$( hex+ ) end:#position {
-        ValueAndSpan {
-            inner: SpannedValue::Integer(i64::from_str_radix(i, 16).unwrap()),
-            span: Span::new(start, end)
-        }
-    }
+pub integer -> SpannedValue = i:$( sign? digit+ ) !("." / ([eE]))
```

Comment:
Note for those who come after me: the `!("." \ ([eE]))` at the end, plus ordering of rules, is how we resolve the issue of `123` being valid EDN `integer` and `float`.